packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -385,4 +389,48 @@ contract MixinExchangeCore is
 
         return fillResults;
     }
+
+   /// @dev Settles an order by transferring assets between counterparties.
+    /// @param order Order struct containing order specifications.
+    /// @param takerAddress Address selling takerAsset and buying makerAsset.
+    /// @param fillResults Amounts to be filled and fees paid by maker and taker.
+    function settleOrder(
```

Comment:
I'm in favour of inlining this function in `fillOrder`. No need to keep it a separate function.